Bio/Restriction/Restriction.py

```diff
@@ -548,11 +547,13 @@ def _drop(cls):
         sequence. Modify the index for site in circular sequences.
         """
         length = len(cls.dna)
-        drop = itertools.dropwhile
-        take = itertools.takewhile
         if cls.dna.is_linear():
-            cls.results = list(drop(lambda x: x <= 1, cls.results))
-            cls.results = list(take(lambda x: x <= length, cls.results))
+
+            def filtering_function(cut_on_watson):
+                cut_on_crick = cut_on_watson - cls.ovhg
+                return all(1 < x <= length for x in (cut_on_watson, cut_on_crick))
+
+            cls.results = list(filter(filtering_function, cls.results))
```

Comment:
A style query, as I didn't immediately remember how filter worked (not indeed the drop and take methods in the older code). See e.g. https://realpython.com/python-filter-function/#replacing-filter-with-a-list-comprehension

```python
# Generating a list with filter()
list(filter(function, iterable))

# Generating a list with a list comprehension
[item for item in iterable if function(item)]
```

How about something like:

```python
def filtering_function(cut_on_watson):
    cut_on_crick = cut_on_watson - cls.ovhg
    return (1 < cut_on_watson <= length) and (1 < cut_on_crick <= length)

cls.results = [cut for cut in cls.results if filtering_function(cut)]
```

Although I feel this can also be improved on...

Reply:
I have made that change